towhee/compiler/utils/callstack.py

```diff
@@ -11,55 +11,138 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import inspect
+from typing import List
+import hashlib
 
 
 class Callstack:
+    def __init__(self):
+        """
+        Initialize a callstack
+        """
 
     def collect(self):
         """
         Collect the frames of current callstack.
         """
-        raise NotImplementedError
+        frames = inspect.stack()
+        frame_list = []
+        for f in frames:
+            frame_list.append(f.frame)
+        return frame_list
```

Comment:
we need a member self._frames to store the callstack, not to recollect the frames at every call.

Reply:
But the stack changes every time we call any function, so we have to recollect the frames every time

Reply:
> we need a member self._frames to store the callstack, not to recollect the frames at every call.

But the stack changes every time we call any function, so we have to recollect the frames every time

Reply:
But the stack changes every time we call any function, so we have to recollect the frames every time @GuoRentong

Reply:
if need store the callstack,  class can be named CallstackSnapshot